src/ex_getln.c

```diff
@@ -974,7 +977,7 @@ cmdline_wildchar_complete(
 		    p_wmnu = 0;
 
 		    // remove match
-		    nextwild(xp, WILD_PREV, 0, escape);
+		    nextwild(xp, WILD_PREV, 0 | (options & ~WIM_NOSELECT), escape);
```

Comment:
This doesn't look right. `WIM_NOSELECT` is `0x10`, which is equal to `WILD_ADD_SLASH`, which is never set in `options`.

Reply:
And the `0 | ` isn't needed.

Reply:
Good catch. It was supposed to be just `nextwild(xp, WILD_PREV, options, escape);`. Don't know how it got in there. `options` is needed here to prevent xp_files from being emptied out if there is only 1 entry. For `noselect` single element completion menu should be retained.

Reply:
Fixed https://github.com/vim/vim/pull/16778